src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2673,7 +2684,21 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
                     }
                 }
 
-                EmitInitObj(type, true, syntaxNode);
+                if (type.IsPointerType() || type.SpecialType == SpecialType.System_UIntPtr)
```

Comment:
That is mostly refactoring of some optimizations to EmitDefaultValue from EmitInitObj.
We had existing tests for some cases and new tests that do something like `default<S1>.Method()` - that is how I noticed that we do some optimizations in one method and some in another - making method's responsibilities unclear.